### ISSUE TITLE
[RFC]: add `@stdlib/array/base/mskfilter-map`

### ISSUE THREAD

**Issue by kgryte** [stdlib-js/stdlib#1322]
### Description

This RFC proposes adding the package `@stdlib/array/base/mskfilter-map`.

The package should be structured similarly to [`@stdlib/array/base/mskfilter`](https://github.com/stdlib-js/stdlib/tree/64931c5fdb4abc3a339b619658d3b5acc089e556/lib/node_modules/%40stdlib/array/base/mskfilter).

The package should expose an API having the same signature as `mskfilter`, with the addition of `clbk` and `thisArg` arguments.

```javascript
function mskfilterMap( x, mask, clbk[, thisArg ] ) {...}
```

The callback function should map each selected value specified by `mask` to the output value.

In short, this package combines `mskfilter` and `map` into a single API call.

Package: `@stdlib/array/base/mskfilter-map`
Alias: `mskfilterMap`

### Related Issues

Related issues #1321 .

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

**Comment by kanhaiya04** [comment]
@kgryte I want to work on this.

**Comment by kgryte** [comment]
Thank you for volunteering to work on this @kanhaiya04. Please feel free to submit a PR implementing this feature.

**Comment by Userf1995** [comment]
sir this issue has been assigned to someone but he has not yet resolved the issue. So can I work on that issue

**Comment by kanhaiya04** [comment]
@Rejoan-Sardar I'm working on it.

**Comment by kgryte** [comment]
@Rejoan-Sardar I suggest finding another issue to work on for the time being.

**Comment by AuenKr** [comment]
Can I work on this issue, as this issue is not assign to anyone currently?

**Comment by kgryte** [comment]
@AuenKr Yes, go for it. Thanks!

**Comment by AuenKr** [comment]
According to my understanding, expected behaviour is output of following code

```javascript
function mskfilterMap( x, mask, clbk[, thisArg ] ) {
	var xget;
	var mget;
	var out;
	var i;
	var currArg;

	if(thisArg!=null) {
		currArg = thisArg;
	}else {
		currArg = null;
	}

	// Resolve accessors for retrieving array elements:
	xget = resolveGetter( x );
	mget = resolveGetter( mask );

	// Extract each desired element from the provided array...
	out = [];
	for ( i = 0; i < x.length; i++ ) {
		if ( mget( mask, i ) ) {
			out.push( clbk.call( currArg, xget( x, i ) ) ); // use `Array#push` to ensure "fast" elements
		}
	}
	return out;
}
```

Is this approach correct, any suggestion to this?
